org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLRename.java

```diff
@@ -0,0 +1,279 @@
+				Position endPosStart = new Position(endPos.getLine(), endPos.getCharacter() - 2);
+				startTagRange = new Range(startPos, startPosEnd);
+				endTagRange = new Range(endPosStart, endPos);
+				return getRenameList(startTagRange, endTagRange, newText);
+			}
+			return Collections.emptyList();
+		}
+		else if(node.isElement()) {
+			DOMElement element = (DOMElement) node;
+			startTagRange = XMLHighlighting.getTagNameRange(TokenType.StartTag, node.getStart(), xmlDocument);
+			endTagRange = element.hasEndTag() ? getTagNameRange(TokenType.EndTag, element.getEndTagOpenOffset(), xmlDocument)
+					: null;
+			if (doesTagCoverPosition(startTagRange, endTagRange, position)) {
+				//Check if xsd namespace rename
+				String fullNodeName = node.getNodeName();
+				int indexOfColon = fullNodeName.indexOf(":");
+				if(indexOfColon > 0) {
+					Position startTagStartPosition = startTagRange.getStart();
+					Position startTagPrefixPosition = new Position(startTagStartPosition.getLine(), startTagStartPosition.getCharacter() + indexOfColon);
+
+					Position endTagStartPosition = endTagRange.getStart();
+					Position endTagPrefixPosition = new Position(endTagStartPosition.getLine(), endTagStartPosition.getCharacter() + indexOfColon);
+
+					Range startTagPrefixRange = new Range(startTagStartPosition, startTagPrefixPosition);
+					Range endTagPrefixRange = new Range(endTagStartPosition, endTagPrefixPosition);
+
+					if(doesTagCoverPosition(startTagPrefixRange, endTagPrefixRange, position)) {// Element prefix rename
+						String prefix = element.getPrefix();
+						return renameElementNamespace(xmlDocument, element, prefix.length(), newText);
+					}
+					else { //suffix rename without wiping namespace
+						String suffixName = element.getLocalName();
+						int suffixLength = suffixName.length();
+						Position startTagEndPosition = startTagRange.getEnd();
+						Position suffixStartPositionStart = new Position(startTagEndPosition.getLine(), startTagEndPosition.getCharacter() - suffixLength);
+
+						Position endTagEndPosition = endTagRange.getEnd();
+						Position suffixEndPositionStart = new Position(endTagEndPosition.getLine(), endTagEndPosition.getCharacter() - suffixLength);
+
+						Range suffixRangeStart = new Range(suffixStartPositionStart, startTagEndPosition);
+						Range suffixRangeEnd = new Range(suffixEndPositionStart, endTagEndPosition);
+
+						return getRenameList(suffixRangeStart, suffixRangeEnd, newText);
+					}
+				}
+				//Regular tag name rename
+				return getRenameList(startTagRange, endTagRange, newText);
+			}
+
+			if(element.equals(xmlDocument.getDocumentElement())) { // If attribute xmlns:ATT_NAME was renamed
+				List<DOMAttr> attributes = element.getAttributeNodes();
+
+				if(attributes == null) {
+					return Collections.emptyList();
+				}
+
+				for (DOMAttr attr : attributes) {
+					DOMNode nameNode = attr.getNodeAttrName();
+
+					if(!attr.getName().startsWith("xmlns")) {
```

Comment:
I think it should b emore clean to have a DOMAttr#isXmlns() method. You can find isXmlNs in the DOMElement which should be moved to DOMAttr